CPP11.md

```diff
@@ -969,6 +970,39 @@ auto a = CountTwos(vec); // 2
 auto b = CountTwos(arr);  // 1
 ```
 
+### regular expressions
+The <regex> library offers tools to handle regular expressions, supporting POSIX and other common syntaxes for regular expressions.   
+
+The std::regex class is initialized with a regular expression and handles parsing the regular expression.   
+std::regex_search() looks for a regex inside a given string, and std::regex_replace() replaces any instances of the regex inside the given string. 
+std::regex_iterator and std::regex_token_iterator allow iteration through matches and submatches (respectively) within the string. 
+
+The following example illustrates the use of regular expressions to identify an email address: 
+
+#include <iostream>
+#include <regex>
+#include <string>
+
+int main() {
+
+    // Define a simple regex pattern for an email address
+    std::regex email_pattern(R"([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})");
+
+    // Input string to check
+    std::string email = "example@example.com";
+
+    // Check if the input string matches the pattern
+    if (std::regex_match(email, email_pattern)) {
+        std::cout << email << " is a valid email address." << std::endl;
+    } else {
+        std::cout << email << " is not a valid email address." << std::endl;
+    }
+
+    return 0;
+}
```

Comment:
For code sample make sure you enclose them in code blocks (using backticks).

Also for brevity, I don't include headers and a detailed implementation using `main` since it distracts from the example. Something like this:

```cpp
// Define a simple regex pattern for an email address.
std::regex email_pattern("[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}");
// Input string to check.
std::string email = "example@example.com";
// Check if the input string matches the pattern:
if (std::regex_match(email, email_pattern)) {
  // matches
} else {
  // no matches
}
```